src/frontends/tensorflow_common/src/op/rank.cpp

```diff
@@ -14,17 +18,35 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 
-ov::OutputVector translate_rank_op(const NodeContext& node) {
-    default_op_checks(node, 1, {"Rank"});
-    auto input = node.get_input(0);
-    auto input_shape = make_shared<v3::ShapeOf>(input, ov::element::i32);
-    auto unsqueeze_input_rank = make_shared<v3::ShapeOf>(input_shape, ov::element::i32);
-    auto input_rank = make_shared<v0::Squeeze>(unsqueeze_input_rank);
+ov::OutputVector translate_rank_op(const NodeContext &node) {
+  default_op_checks(node, 1, {"Rank"}, true);
+  auto input = node.get_input(0);
+  auto complex_type_mark =
+      as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+  auto input_shape = make_shared<v3::ShapeOf>(input, ov::element::i32);
```

Comment:
please move this line under `if` below.
You need to compute separate `input_shape` inside `if` like to avoid compute shape for `ComplexTypeMark`:

```
input = complex_type_mark->input_value(0);
auto input_shape = make_shared<v3::ShapeOf>(input, ov::element::i32);
```

Reply:
Thank you very much! It solved the BUG before.